### PR TITLE
enable merkle skip verification for admins in prod

### PR DESCRIPTION
**Pull request by maxtaco**
- pin to first live site commit @835903
- should have no effect on dev

### CODE REVIEW

File: go/libkb/config.go [hunk 1]
```diff
@@ -414,11 +414,17 @@ func (f JSONConfigFile) GetGpgOptions() []string {
 func (f JSONConfigFile) GetRunMode() (RunMode, error) {
 	var err error
 	var ret RunMode = NoRunMode
-	if s, isSet := f.GetStringAtPath("run-mode"); isSet {
+	if s, isSet := f.GetStringAtPath("run_mode"); isSet {
```
Comment (oconnor663) [inline go/libkb/config.go]: Should we change `chat-db` above too? Also any chance this is going to break someone's dev setup?

File: go/libkb/constants.go [hunk 2]
```diff
@@ -484,4 +484,4 @@ const (
 )
 
 // Eventually, this will be set to the first merkle root block with skip pointers.
```
Comment (oconnor663) [inline go/libkb/constants.go]: Update this comment?

File: go/libkb/features.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import (
+	"strings"
+)
+
+type Feature string
+type FeatureFlags []Feature
+
+// StringToFeatureFlags returns a set of feature flags
+func StringToFeatureFlags(s string) (ret FeatureFlags) {
+	s = strings.TrimSpace(s)
```
Comment (oconnor663) [inline go/libkb/features.go]: Should we trim space after splitting instead?

File: go/libkb/features.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	return ret
+}
+
+// Admin returns true if the admin feature set is on
+func (set FeatureFlags) Admin() bool {
+	for _, f := range set {
+		if f == Feature("admin") {
+			return true
+		}
+	}
```
Comment (oconnor663) [inline go/libkb/features.go]: Maybe make it a `map[string]struct{}` to get rid of these loops?

File: go/libkb/merkle_client.go [hunk 5]
```diff
@@ -685,13 +688,17 @@ func (mc *MerkleClient) LastRoot() *MerkleRoot {
 }
 
 func (mc *MerkleClient) FirstSeqnoWithSkips() *Seqno {
+
 	if mc.G().Env.GetRunMode() == ProductionRunMode {
-		return FirstProdMerkleSeqnoWithSkips
+		if mc.G().Env.GetFeatureFlags().Admin() {
+			return &FirstProdMerkleSeqnoWithSkips
```
Comment (oconnor663) [inline go/libkb/merkle_client.go]: What should happen if I'm in prod mode but I set `--server=http://localhost:3000` or something like that? Is this a configuration that was working before but that won't work after this?

**Comment by oconnor663** [comment]
Sounds like "prod mode on some server other than prod" is totally unsupported, so ignore that one :)

LGTM, your call on the smaller changes.

File: go/libkb/config.go (hunk 1 above)
Comment (maxtaco) [inline go/libkb/config.go]: will change both.  i doubt this will break anyone

File: go/libkb/merkle_client.go (hunk 5 above)
Comment (maxtaco) [inline go/libkb/merkle_client.go]: yeah, i don't think that ever worked previously

**Pull request merged**